ansible_collections/arista/avd/plugins/action/eos_designs_facts.py

```diff
@@ -99,6 +109,44 @@ def create_avd_switch_facts_instances(self, fabric_hosts: list, hostvars: object
         default_vars : dict
             Default variables shared between all fabric_hosts
 
+        Returns
+        -------
+        dict
+            hostname1 : dict | None
+            hostname2 : dict | None
+            ...
+        bool
+            Did validation fail for one or more of the hosts
+        """
+        # Load schema tools once with empty host.
+        avdschematools = AvdSchemaTools(self._schema, "", display, self._conversion_mode, self._validation_mode)
+
+        all_hostvars = {}
+        failed = False
+        for host in fabric_hosts:
+            # Using ChainMap to avoid copying data between defaults, hostvars and local template_vars
+            all_hostvars[host] = ChainMap({}, hostvars.get(host), default_vars)
+
+            # Set correct hostname and perform conversion and validation
```

Comment:
What does `set correct hostname` mean here?

Reply:
A line was lost in rebasing, but I have added it now.